code/datums/status_effects/debuffs.dm

```diff
@@ -519,9 +521,13 @@
 
 /// Resisting the debuff will allow the debuff's owner to remove some stacks from themselves.
 /datum/status_effect/stacking/intoxicated/proc/resist_debuff()
-	if(!do_after(debuff_owner, 3 SECONDS, TRUE, debuff_owner, BUSY_ICON_GENERIC))
+	is_resisting = TRUE
```

Comment:
Never gets set to false. Also use the `do_actions` list on the owner mob instead of a var. (`length(owner.do_actions)`, etc)

Reply:
i somehow also managed to forget to add a check for returning at all, good job me